src/ConcreteOperations/volume.jl

```diff
@@ -0,0 +1,34 @@
+export volume
+
+"""
+    volume(H::AbstractHyperrectangle)
+
+Return the volume of a hyperrectangular set.
+
+### Input
+
+- `H` -- hyperrectangular set
+
+### Output
+
+The (Euclidean) volume of ``H``.
+
+### Algorithm
+
+The volume of the hyperrectangle ``H`` with vector radius ``r`` is
+``2ⁿ ∏ᵢ rᵢ`` where ``rᵢ`` denotes the ``i``-th component of ``r``.
+"""
+function volume(H::AbstractHyperrectangle{N}) where {N<:AbstractFloat}
+    r = radius_hyperrectangle(H)
+    n = dim(H)
+    α = exp(n*log(2))
+    vol = α * prod(r)
+    return vol
+end
+
+# fallback for rational
+function volume(H::AbstractHyperrectangle{N}) where {N}
+    r = radius_hyperrectangle(H)
+    vol = prod(2 .* r)
+    return vol
+end
```

Comment:
The following implementations are more efficient because they only create one or zero intermediate vectors.
```julia
function volume2(H::AbstractHyperrectangle{N}) where {N<:Real}
    vol = one(N)
    for ri in radius_hyperrectangle(H)
        vol *= 2 * ri
    end
    return vol
end

function volume3(H::AbstractHyperrectangle{N}) where {N<:Real}
    vol = one(N)
    for i in 1:dim(H)
        vol *= 2 * radius_hyperrectangle(H, i)
    end
    return vol
end

julia> B = BallInf(zeros(Rational{Int}, 1000), 1//2);

julia> @btime volume($B)
  96.468 μs (2 allocations: 31.50 KiB)
1//1

julia> @btime volume2($B)
  94.259 μs (1 allocation: 15.75 KiB)
1//1

julia> @btime volume3($B)
  92.144 μs (0 allocations: 0 bytes)
1//1
```

Reply:
Thanks for your comments. I didn't consider optimizing for BallInf, since the application that i had in mind requires to use Hyperrectangles (splitting a BallInf by the longer direction is no longer a box). For hyperrectangles, the functions that i added do not allocate.

After some tests, the implementation that works better for hyperrectangles is the variation `mapreduce(x -> 2x, *, radius_hyperrectangle(H))`. For boxes, your `volume3` is faster for dim approx smaller than 16, eventually there is a crossover with the exponentiation function (`vol = exp(n*log(2r))`) as it has constant cost. For n=100, the exponential is 60% faster than volume3. I would tend to use volume3 however, since n > 20 seems "big" for this type of computation..

Reply:
> I didn't consider optimizing for BallInf, since the application that i had in mind requires to use Hyperrectangles (splitting a BallInf by the longer direction is no longer a box). For hyperrectangles, the functions that i added do not allocate.

Oh, true, I forgot that. Whatever you prefer, then.

Reply:
> I would tend to use volume3 however, since n > 20 seems "big" for this type of computation.

What about implementing both and choosing the best implementation based on `dim(H)`?

Reply:
:ok_hand:  BTW the experiments that i did are in [this notebook](https://nbviewer.jupyter.org/github/mforets/escritoire/blob/master/sets/Volume.ipynb)